sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -286,8 +286,15 @@ protected virtual async Task OnMessageHandler(ServiceBusReceivedMessage message,
                     ServiceBusEventSource.Log.ProcessorRenewMessageLockStart(_identifier, 1, message.LockToken);
                     TimeSpan delay = CalculateRenewDelay(message.LockedUntil);
 
-                    await Task.Delay(delay, cancellationToken).ConfigureAwait(false);
-                    if (Receiver.IsClosed)
+                    // We're awaiting the task created by 'ContinueWith' to avoid awaiting the Delay task which may be canceled
```

Comment:
Why not just add an explicit `try/catch` and ignore the cancellation exception?   To me, that would be more clear on the intent.

Reply:
The goal of the change is to prevent exceptions from firing at all.

Reply:
I'm not sure this does that... I believe it just buries the exception as part of the continuation.   No?

Reply:
The TaskCanceledException doesn't end up getting thrown when using ContinueWith. I've verified this locally with the Diagnostic Tools.

Reply:
Interesting.   I've been trying to trace through the .NET source but haven't found the spot where it normally gets thrown yet.   I'll take your word for it and be grateful that I learned something today.  😄

Reply:
It throws somewhere around https://source.dot.net/#System.Private.CoreLib/TaskAwaiter.cs,181

Reply:
But if we are chaining tasks the last task succeeds even though the previous ones are marked as failed. Because the ContinueWith doesn't check for failures.